pandas/core/common.py

```diff
@@ -2072,7 +2072,7 @@ def _random_state(state=None):
     elif isinstance(state, np.random.RandomState):
         return state
     elif state is None:
-        return np.random.RandomState()
+        return np.random.mtrand._rand
```

Comment:
We want results to be reproducible after calling `np.random.seed()`. But we don't need to reuse exactly the same seed -- it's OK to also use a seed derived from NumPy's seed.

Using the exact same seed would only be necessary if we want to promise the `sample` makes the exact same choice as `np.random.choice`. But that's really an implementation detail.

(Clearly a comment in the code would also be in order if we use my suggested approach.)